File: app/src/main/java/com/indiewalk/watchdog/earthquake/net/EarthquakeFirebaseJobService.java
package com.indiewalk.watchdog.earthquake.net;

import android.util.Log;

import com.firebase.jobdispatcher.JobParameters;
import com.firebase.jobdispatcher.JobService;
import com.indiewalk.watchdog.earthquake.SingletonProvider;


public class EarthquakeFirebaseJobService extends JobService {
    private static final String TAG = EarthquakeFirebaseJobService.class.getSimpleName();


    /**
     * ---------------------------------------------------------------------------------------------
     * Job entry point, called by the Job Dispatcher.
     * @return  whether there is more work remaining.
     * ---------------------------------------------------------------------------------------------
     */
    @Override
    public boolean onStartJob(final JobParameters jobParameters) {
        Log.d(TAG, "Earthquake remote fetching Job service started");

        EarthquakeNetworkDataSource networkDataSource =
                ((SingletonProvider)SingletonProvider.getsContext()).getNetworkDatasource();


        networkDataSource.startFetchEarthquakeService();

        jobFinished(jobParameters, false);
        return true;
    }


    @Override
    public boolean onStopJob(JobParameters jobParameters) {
        return false;
    }
}

File: app/src/main/java/com/indiewalk/watchdog/earthquake/net/EarthquakeFirebaseJobService.kt
package com.indiewalk.watchdog.earthquake.net

import android.util.Log

import com.firebase.jobdispatcher.JobParameters
import com.firebase.jobdispatcher.JobService
import com.indiewalk.watchdog.earthquake.SingletonProvider


class EarthquakeFirebaseJobService : JobService() {


    /**
     * ---------------------------------------------------------------------------------------------
     * Job entry point, called by the Job Dispatcher.
     * @return  whether there is more work remaining.
     * ---------------------------------------------------------------------------------------------
     */
    override fun onStartJob(jobParameters: JobParameters): Boolean {
        Log.d(TAG, "Earthquake remote fetching Job service started")

        val networkDataSource = (SingletonProvider.getsContext() as SingletonProvider).networkDatasource


        networkDataSource?.startFetchEarthquakeService()

        jobFinished(jobParameters, false)
        return true
    }


    override fun onStopJob(jobParameters: JobParameters): Boolean {
        return false
    }

    companion object {
        private val TAG = EarthquakeFirebaseJobService::class.java.simpleName
    }
}
